examples/decoding/plot_decoding_xdawn_eeg.py

```diff
@@ -62,30 +61,28 @@
                 add_eeg_ref=False, verbose=False)
 
 # Create classification pipeline
-clf = make_pipeline(Xdawn(n_components=3),
-                    EpochsVectorizer(),
+X, labels = EpochsVectorizer().fit_transform(epochs)
+clf = make_pipeline(XdawnTransformer(n_components=3,
+                    n_chan=epochs.info['nchan']),
                     MinMaxScaler(),
                     LogisticRegression(penalty='l1'))
 
-# Get the labels
-labels = epochs.events[:, -1]
-
 # Cross validator
 cv = StratifiedKFold(y=labels, n_folds=10, shuffle=True, random_state=42)
 
 # Do cross-validation
-preds = np.empty(len(labels))
+y_preds = np.empty(len(labels))
```

Comment:
note that this is particularly bad practice for `y_preds` because the cv may not be exhaustive and may thus be filled with weird values.